TVPaint JSON export (Experimental feature)

shun-iwasawa opened the pull request:
This PR will enable to export the xsheet information in [TVPaint JSON file format](https://www.tvpaint.com/doc/tvp11/index.php?id=lesson-import-export-export-files-layers-structure-json&lang=en).

The TVP-JSON file can be loaded not only in TVPaint itself, but also in AfterEffects by using the AE script written and distributed by courtesy of Clément Berthaud for TVPaint Developpement:

> Title: Import TVPaint Animation Clip Structure to After Effects
> Author: Clément Berthaud for TVPaint Developpement
> Version: 5.7.1
> License: [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/)
> 
> [AfterEffects Import TVPaint JSON.zip](https://github.com/opentoonz/opentoonz/files/6875066/AfterEffects.Import.TVPaint.JSON.zip)

The export command is placed under `Menu bar > File > Export` submenu.
Since this PR is experimental, the feature is very limited as of now:
- The level file must be placed at the same or child folder relative to the scene file.
- Only the columns containing raster levels can be exported.
- Exporting camera work information is still missing.
- Import feature is still missing.

RodneyBaker commented:
@shun-iwasawa 
I cannot test with TVPaint or After Effects at this time but can confirm the successful export of data into the JSON file.
The main issue I ran into was that I had not saved my scene file and therefore the export process defaulted to the standard error message of 'cannot export columns'.  

I did also validate that only raster levels can be used at this time and that the levels need to be close to the scene file (I mostly tested with levels inside the same directory as the scene file as I knew that should work.

It is exciting to see this experimental feature and I look forward to its future development, perhaps especially the ability to import the JSON files back into Opentoonz.

I will defer to you for merging as I lack the capability to test the whole process but... the initial export out of Opentoonz  looks good.

shun-iwasawa commented:
Fixed a bug that the stacking order of the layers are inverted in the exported file.

@RodneyBaker 
Thank you for testing the PR!

> The main issue I ran into was that I had not saved my scene file and therefore the export process defaulted to the standard error message of 'cannot export columns'.

Untitled scene cannot be exported because TVP-JSON file stores the level file paths relative to the scene file. 
I modified the warning popup to show more detailed instructions.

gab3d commented:
Great work! :muscle: 
Json export would also be very useful to be able to export rigged character animation cycles to game engines, as other programs like Spine do.

shun-iwasawa commented:
This feature was originally requested from some Japanese animation studio. I've asked them for testing this PR and they confirmed that it works OK.
I'll merge this. Any feedback would be appreciated!
